src/secure-proxy-handler.ts

```diff
@@ -92,6 +94,22 @@ function copySecureOwnDescriptors(env: SecureEnvironment, shadowTarget: SecureSh
     }
 }
 
+function callWithErrorBoundaryProtection(env: SecureEnvironment, fn: () => RawValue): RawValue {
+    let result;
+    try {
+        result = fn();
+    } catch (rawError) {
+        // This error occurred when a sandbox attempted to invoke a function,
+        // setter, getter, or any other operation on the outer realm.
+        // TODO: what should we do with this error?
+        //       by default, it will be just proxy it, but the stack and everything
+        //       from the original error will be accessible into the sandbox, although
+        //       no object identity will be leaked.
+        throw rawError;
```

Comment:
@jdalton this is the area of research at this point. I will add more notes into the original issue #8, to see if we can define what needs to be done here.